### PR TITLE
zh: update feature-shard-merge-optimistic.md

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What is changed, added or deleted? (Required)
- move restriction to the beginning
- add unsupported sql statements

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by GMHDBJD** [comment]
@csuzhangxc @ireneontheway PTAL

File: zh/feature-shard-merge-optimistic.md [hunk 1]
```diff
@@ -146,24 +172,6 @@ ALTER TABLE `tbl00` ADD COLUMN `Age` INT DEFAULT -1;
...
-
-此外，不论是使用“乐观协调”或“悲观协调”，DM 仍是有以下限制：
-
-- 增量同步任务需要确保开始同步的 binlog position 对应的各分表的表结构必须一致。
-- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止 `CREATE/RENAME TABLE`）。
-- 不支持 `DROP TABLE` / `DROP DATABASE`。
-- TiDB 不支持的 DDL 语句在 DM 也不支持。
-- 新增列的默认值不能包含 `current_timestamp`、`rand()`、`uuid()` 等，否则会造成上下游数据不一致。
-
 ## 乐观协调模式的配置
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:175]: 把这个也挪到前面去算了吗？

Comment (csuzhangxc) [reply zh/feature-shard-merge-optimistic.md:175]: 还包括 "使数据不一致的操作"

File: zh/feature-shard-merge-optimistic.md [hunk 2]
```diff
@@ -18,6 +18,75 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
...
+
+```SQL 
+ALTER TABLE `tbl00` ADD COLUMN `Age` INT DEFAULT -1;
+```
+
+![optimistic-ddl-fail-example-3](/media/optimistic-ddl-fail-example-3.png)
+
+此时所有来自 `tbl00` 的 `Age` 都不一致了。这是由于 `DEFAULT 0` 和 `DEFAULT -1` 互不兼容。虽然 DM 遇到这种情况会报错，但上下游不一致的问题就需要手动去解决。
+
+### 使数据不一致的操作
```
Comment (csuzhangxc) [inline zh/feature-shard-merge-optimistic.md:79]: 把 "使数据不一致的操作" 放到 "例子" 之前？

Comment (GMHDBJD) [reply zh/feature-shard-merge-optimistic.md]: 改了 ebdf8ea

File: zh/feature-shard-merge-optimistic.md [hunk 3]
```diff
@@ -18,6 +18,74 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
...
+- `ALTER TABLE table_name ADD COLUMN column_name datatype NOT NULL`（添加无默认值的 not null 的列）。
+- `ALTER TABLE table_name ADD COLUMN column_name datetime DEFAULT NOW()`（增加的列默认值不固定）。
+- `ALTER TABLE table_name RENAME COLUMN column_1 TO column_2;`（重命名列）。
+- `ALTER TABLE table_name RENAME INDEX index_1 TO index_2;`（重命名索引）。
+
+此外，不论是使用“乐观协调”或“悲观协调”，DM 仍是有以下限制：
+
+- 增量同步任务需要确保开始同步的 binlog position 对应的各分表的表结构必须一致。
+- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止 `CREATE/RENAME TABLE`）。
+- 不支持 `DROP TABLE` / `DROP DATABASE`。
```
Comment (ireneontheway) [inline zh/feature-shard-merge-optimistic.md:46]:
```suggestion
- 不支持 `DROP TABLE`/`DROP DATABASE`。
```
> "斜杠的两旁禁止加空格。" from [中文技术文档规范
](https://docs.google.com/document/d/1b6ZhZD33OoM8AacpKksGGSuxJWReLkNnSt8eSc1kTXc/edit#)

File: zh/feature-shard-merge-optimistic.md [hunk 4]
```diff
@@ -18,6 +18,74 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
...
+
+在任务的配置文件中指定 `shard-mode` 为 `optimistic` 则使用“乐观协调”模式，示例配置文件可以参考 [DM 任务完整配置文件介绍](task-configuration-file-full.md)。
+
+## 使用限制
+
+使用“乐观协调”模式有一定的风险，需要严格遵照以下方针：
+
+- 执行每个批次的 DDL 前和后，要确保每个分表的结构达成一致。
+- 进行灰度 DDL 时，只集中在一个分表上测试。
+- 灰度完成后，在其他分表上尽量以最简单直接的 DDL 迁移到最终的 schema，而不要重放灰度测试中对或错的每一步。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:31]:
```suggestion
- 灰度完成后，在其他分表上尽量以最简单直接的 DDL 迁移到最终的 schema，而不要重新执行灰度测试中对或错的每一步。
```

File: zh/feature-shard-merge-optimistic.md [hunk 5]
```diff
@@ -18,6 +18,74 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
...
+
+- `ALTER TABLE table_name ADD COLUMN column_name datatype NOT NULL`（添加无默认值的 not null 的列）。
+- `ALTER TABLE table_name ADD COLUMN column_name datetime DEFAULT NOW()`（增加的列默认值不固定）。
+- `ALTER TABLE table_name RENAME COLUMN column_1 TO column_2;`（重命名列）。
+- `ALTER TABLE table_name RENAME INDEX index_1 TO index_2;`（重命名索引）。
+
+此外，不论是使用“乐观协调”或“悲观协调”，DM 仍是有以下限制：
+
+- 增量同步任务需要确保开始同步的 binlog position 对应的各分表的表结构必须一致。
+- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止 `CREATE/RENAME TABLE`）。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:45]:
```suggestion
- 进入 sharding group 的新表必须与其他成员的表结构一致（正在执行一个 DDL 批次时禁止执行 `CREATE/RENAME TABLE` 命令）。
```

File: zh/feature-shard-merge-optimistic.md [hunk 6]
```diff
@@ -18,6 +18,74 @@ DM 支持在线上执行分库分表的 DDL 语句（通称 Sharding DDL），
...
+- 不支持 `DROP TABLE`/`DROP DATABASE`。
```
Comment (yikeke) [inline zh/feature-shard-merge-optimistic.md:46]:
```suggestion
- 不支持 `DROP TABLE`/`DROP DATABASE` 命令。
```

**Pull request merged**